### ISSUE TITLE
Game crashes when starting specific songs - deleting ~/.cache/performous temporarily resolves the problem

### ISSUE THREAD

**Issue by BWagener** [performous/performous#977]
### Performous & operating system versions

Performous: 1.3.1
Operating system: Arch Linux on the Steam Deck

### What is the problem?

The game crashes when starting specific songs - deleting ~/.cache/performous temporarily resolves the problem.

The error message:
FATAL ERROR: cannot create std::vector larger than max_size()
core/error: cannot create std::vector larger than max_size()

![image](https://github.com/performous/performous/assets/19321042/8325ca2c-3eec-4c72-828c-e3b8c43e4244)

It occurs when selecting a song to play. It only happens to a few specific songs.
When I delete the game's cache directory in ~/.cache I can play the affected songs again.

After a while, the problem returns for the same songs again. Playing other songs keeps working fine.

### Steps to reproduce

Play the game regularly and try out different songs, after a while it starts crashing for some songs.

### Additional context

Let me know if there's anything I can do such as run specific debug versions etc.

### Log(s) & preferences file(s)

This is the crash log of the most recent occurence:

```
songs/notice: Done Loading. Loaded 18592 Songs.
songs/notice: Done Caching.
image/error: Writing PNG failed
image/error: Writing PNG failed
stderr/info: libpng warning: Image height is zero in IHDR
ffmpeg/warning: Negative audio sample_position -2208 seconds, frame ignored.
ffmpeg/warning: Negative audio sample_position -2114 seconds, frame ignored.
ffmpeg/warning: Negative audio sample_position -2208 seconds, frame ignored.
ffmpeg/warning: Negative audio sample_position -2114 seconds, frame ignored.
ffmpeg/warning: Negative audio sample_position -2208 seconds, frame ignored.
ffmpeg/warning: Negative audio sample_position -2114 seconds, frame ignored.
ffmpeg/warning: Negative audio sample_position -2404 seconds, frame ignored.
ffmpeg/warning: Negative audio sample_position -2336 seconds, frame ignored.
ffmpeg/warning: Negative audio sample_position -2404 seconds, frame ignored.
ffmpeg/warning: Negative audio sample_position -2336 seconds, frame ignored.
core/notice: More details might be available in "/home/deck/.cache/performous/infolog.txt".
stderr/notice: 8 messages logged to stderr/info
FATAL ERROR: cannot create std::vector larger than max_size()
core/error: cannot create std::vector larger than max_size()
If you think this is a bug in Performous, please report it at 
  https://github.com/performous/performous/issues
```
[config.xml.txt](https://github.com/performous/performous/files/14735635/config.xml.txt)
[infolog.txt](https://github.com/performous/performous/files/14735636/infolog.txt)
[cache.zip](https://github.com/performous/performous/files/14735638/cache.zip)

**Comment by twollgam** [comment]
I looked at the provided data but did not found which song data causes the problem. @BWagener can you get the information which song causes the problem.

**Comment by BWagener** [comment]
Of course, I'll dm you some examples of songs I encountered this issue with when I can.
I spotted it with a couple of songs which have been working fine for a long time in previous versions.

**Comment by BWagener** [comment]
I'm keeping track of three songs I was currently facing this problem with.
If I delete just those songs from the songs.json file in the cache I can play them again.

Something all three songs have in common is I have each song twice, from two different sources - both versions work until the problem occurs, then one stops working (so far it's always been the versions I sing).

I deleted the entries for the three songs which had the problem in the songs.json file, reran the game and closed it. Took a copy of the songs.json file and will compare the copy to the latest version in cache once the problem reoccurs.

**Comment by BWagener** [comment]
Okay, so results of this approach are as follows:

- any broken song has loadStatus: 2 in the songs.json file.
- any song that's still working has loadStatus: 1
- after singing a song and exiting the game as usual the songs entry in the songs.json file is updated from loadStatus: 1 to loadStatus: 2
- I can "break" any song by changing its loadStatus value to 2 in the songs.json file, the same error as in the screenshot in the issue shows up from then on

**Comment by Baklap4** [comment]
So i think this is due to the fact that we calculate the duration which can overflow sometimes, but i don't quite know why. I thought this was fixed by the "detect faulty/no mp3" pr but sadly no :(

Can you try run with a debugger and collect a stacktrace?

**Comment by BWagener** [comment]
I'd need to have a development setup for this right? I haven't got that set up.

Not sure if this is how you can test it, but could you run it with a debugger and just set any song's loadStatus to 2 beforehand and then launch that song?

**Comment by Baklap4** [comment]
> I'd need to have a development setup for this right? I haven't got that set up.

That's correct, sorry for the assumption hahah

I'll give it a go with the suggestion and post results later 👍 
Is this problem still occuring on latest master?

**Comment by BWagener** [comment]
Nice.

I'll test if it also occurs on latest master next week, as I'm on holiday.

**Comment by BWagener** [comment]
@Baklap4 should I close this issue?

It seems to be addressed in master.

One question I still have is what causes the loadStatus to be updated to 2 after playing these songs?

**Comment by Baklap4** [comment]
Just tested it locally the problem still seems to exist, but now that we have a repro. We can try to fix it, thanks for the info!
